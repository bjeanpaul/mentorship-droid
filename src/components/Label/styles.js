import { StyleSheet } from 'react-native';

import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import colors from 'src/constants/colors';

export default StyleSheet.create({
  label: {
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
    fontSize: 14,
    textAlign: 'left',
    color: colors.LABEL_TEXT,
    marginBottom: 9,
  },
});
