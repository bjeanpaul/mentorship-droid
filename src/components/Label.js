import React, { PropTypes } from 'react';
import { StyleSheet } from 'react-native';

import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import Text from './Text';

const styles = StyleSheet.create({
  label: {
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
    fontSize: 14,
    textAlign: 'left',
    color: '#9fb1b3',
    marginBottom: 9,
  },
});

const Label = ({ title, style }) =>
  <Text style={[styles.label, style]}>{title}</Text>;

Label.propTypes = {
  title: PropTypes.string.isRequired,
  style: PropTypes.any,
};
export default Label;
