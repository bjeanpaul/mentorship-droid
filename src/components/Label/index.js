import React, { PropTypes } from 'react';
import { StyleSheet } from 'react-native';

import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import colors from 'src/constants/colors';
import Text from './Text';

const styles = StyleSheet.create({
  label: {
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
    fontSize: 14,
    textAlign: 'left',
    color: colors.LABEL_TEXT,
    marginBottom: 9,
  },
});

const Label = ({ title, style }) =>
  <Text style={[styles.label, style]}>{title}</Text>;

Label.propTypes = {
  title: PropTypes.string.isRequired,
  style: PropTypes.any,
};
export default Label;
