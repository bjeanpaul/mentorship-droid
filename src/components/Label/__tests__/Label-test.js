import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import Label from 'src/components/Label';


describe('Label', () => {
  it('should render its title', () => {
    const el = shallow(<Label title="foo" />);
    expect(el.children().text()).toEqual("foo")
  });

  fit('should support style overriding', () => {
    const el = shallow(
      <Label
        title="foo"
        style={{fontSize: 23}}
      />);

      expect(el.props()).toEqual(jasmine.objectContaining({
        style: jasmine.arrayContaining([{fontSize: 23}])
      }))
  });
});
