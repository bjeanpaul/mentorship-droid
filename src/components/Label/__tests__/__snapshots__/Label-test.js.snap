exports[`Label should render its title 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center",
      },
      Array [
        Object {
          "color": "#9fb1b3",
          "fontFamily": "Brandon Text_bold",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginBottom": 9,
          "textAlign": "left",
        },
        undefined,
      ],
    ]
  }>
  foo
</Text>
`;

exports[`Label should support style overriding 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center",
      },
      Array [
        Object {
          "color": "#9fb1b3",
          "fontFamily": "Brandon Text_bold",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginBottom": 9,
          "textAlign": "left",
        },
        Object {
          "fontSize": 23,
        },
      ],
    ]
  }>
  foo
</Text>
`;
